components/SearchInput.vue

```diff
@@ -0,0 +1,179 @@
+<template>
+  <div
+    class="search"
+    aria-label="search"
+    role="search"
+    :class="{ 'is-active': isActive }"
+  >
+    <div class="field">
+      <div class="control has-icons-left has-icons-right">
+        <input
+          :id="id"
+          ref="input"
+          v-model="searchValue"
+          v-focus="focus"
+          class="input"
+          name="search"
+          type="text"
+          placeholder="Search..."
+          @keyup.exact="onInput"
+          @keyup.enter="onEnter"
+        >
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { isEmpty } from 'lodash'
+
+export default {
+  name: 'HcSearchInput',
+  props: {
+    id: {
+      type: String,
+      default: 'nav-search'
+    },
+    value: {
+      type: String,
+      default: ''
+    },
+    delay: {
+      type: Number,
+      default: 700
+    },
+    focus: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data() {
+    return {
+      searchValue: '',
+      searchProcess: null,
+      searching: false
+    }
+  },
+  computed: {
+    isActive() {
+      return !isEmpty(this.searchValue)
+    }
+  },
+  watch: {
+    value(value) {
+      this.$nextTick(() => {
+        this.updateValue()
+      })
+    }
+  },
+  mounted() {
+    this.updateValue()
+  },
+  methods: {
+    updateValue() {
+      if (!this.value) {
+        this.searchValue = ''
+      } else if (this.value.toString() !== this.searchValue.toString()) {
+        this.searchValue = this.value.toString()
+      }
+    },
+    onInput() {
+      clearTimeout(this.searchProcess)
+      this.searching = true
+      // skip on less then three letters
+      if (this.searchValue && this.searchValue.toString().length < 3) {
+        return
+      }
+      // skip if nothing changed
+      if (this.searchValue === this.value) {
+        return
+      }
+      this.searchProcess = setTimeout(() => {
+        this.searching = false
+        this.$emit('search', this.searchValue.toString())
+      }, this.delay)
+    },
+    onEnter() {
+      clearTimeout(this.searchProcess)
+      this.$nextTick(() => {
+        clearTimeout(this.searchProcess)
+      })
+      this.searching = false
+      this.$emit('search', this.searchValue.toString())
+    },
+    clear() {
+      clearTimeout(this.searchProcess)
+      this.searching = false
+      this.searchValue = ''
+      if (this.value !== this.searchValue) {
+        this.$emit('search', '')
+      }
+    }
+  }
+}
```

Comment:
Nice, could you provide a component test for this? You can use [`vue-test-utils`](https://vue-test-utils.vuejs.org/api/) and get inspiration from [component tests in our legacy repository](
https://github.com/Human-Connection/WebApp/blob/blacklist_feature/components/Global/Elements/BlockButton/BlockButton.test.js).

Reply:
yes, writing them as we speak... far from finished